Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/HDInsightNewAPI/ClusterOperationNewAPIImpl.java

```diff
@@ -0,0 +1,156 @@
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.sdk.cluster.HDInsightNewAPI;
+
+import com.microsoft.azure.hdinsight.common.logger.ILogger;
+import com.microsoft.azure.hdinsight.sdk.cluster.ClusterOperationImpl;
+import com.microsoft.azure.hdinsight.sdk.common.AzureManagementHttpObservable;
+import com.microsoft.azure.hdinsight.sdk.common.errorresponse.ForbiddenHttpErrorStatus;
+import com.microsoft.azuretools.authmanage.AuthMethodManager;
+import com.microsoft.azuretools.authmanage.models.SubscriptionDetail;
+import com.microsoft.azuretools.azurecommons.helpers.AzureCmdException;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.entity.StringEntity;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+
+public class ClusterOperationNewAPIImpl extends ClusterOperationImpl implements ILogger {
+    private static final String VERSION = "2015-03-01-preview";
+    private HDInsightUserRoleType roleType;
+
+    public boolean isProbeGetConfigurationSucceed(final SubscriptionDetail subscription, final String clusterId) {
+        try {
+            String managementURI = AuthMethodManager.getInstance().getAzureManager().getManagementURI();
+            AzureManagementHttpObservable httpObservable = new AzureManagementHttpObservable(subscription, VERSION);
+            String url = URI.create(managementURI)
+                    .resolve(clusterId.replaceAll("/+$", "") + "/configurations").toString();
+            StringEntity entity = new StringEntity("", StandardCharsets.UTF_8);
+            entity.setContentType("application/json");
+            httpObservable
+                    .withUuidUserAgent()
+                    .post(url, entity, null, null, ClusterConfiguration.class)
+                    .toBlocking()
+                    .singleOrDefault(null);
+            setRoleType(HDInsightUserRoleType.OWNER);
+            return true;
+        } catch (RuntimeException ex) {
```

Comment:
Could you catch `ForbiddenHttpErrorStatus` directly?

Reply:
Exception will be wrapped to `RuntimeException` for `BlockingObservable`. So we can't catch `ForbiddenHttpErrorStatus` directly.

---
In reply to: [271121944](https://github.com/Microsoft/azure-tools-for-java/pull/2964#discussion_r271121944) [](ancestors = 271121944)

Reply:
You may try `onErrorReturn()` before `toBlocking()`